src/keycloak/chart/templates/alb-envoyfilter.yaml

```
@@ -0,0 +1,63 @@
+# Copyright 2025 Defense Unicorns
+# SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+
+{{- if eq .Values.thirdPartyIntegration.tls.tlsTermination "external" }}
+{{- if .Values.thirdPartyIntegration.tls.tlsCertificateHeader | default "" | ne "" }}
```

In the name of simplicity I think we could just make this:
```suggestion
{{- if .Values.thirdPartyIntegration.tls.tlsCertificateHeader }}
```

And then not even have `thirdPartyIntegration.tls.tlsTermination` as a value? But if that seems too magical we could have both and do something like:
```suggestion
{{- if and (eq .Values.thirdPartyIntegration.tls.tlsTermination "external") .Values.thirdPartyIntegration.tls.tlsCertificateHeader }}
```

I'm just unsure right now if there's a use-case for:
- setting external termination, but not setting a TLS cert header
- setting default termination, and setting a TLS cert header

With the current templating both of these scenarios would essentially just deploy the "defaults" (no envoyfilter, normal headers, etc).

---

I think there is - you might want to terminate TLS on the ALB level and just don't use x509 auth (CAC) at all. It's still a valid use case. Having said that, I'll revert your suggestion as someone may actually try to deploy it that way.

---

After thinking for a while, I had to revert all these changes. 

It's a perfectly valid use case to terminate the TLS at the edge and just don't forward Client Certificate to Istio. In this case, we just need to have x509 auth disabled (which has a corresponding configuration option).

Technically, this means, that:

* `tlsCertificateHeader` can be empty and we want to apply defaults in that case. 
* The `tlsTermination` should be set to external to activate the EnvoyFilter to do HTTP Client Cert Header modifications.

---

I guess my observation is just that `tls.tlsTermination` set to `external` on its own has no effect. It's part of the envoyfilter conditional but it doesn't matter if you aren't also setting `tls.tlsCertificateHeader`. _Right now_ it just feels like an extra value to set with no purpose. 

If we envision using it to control other features going forward that's fine - I just want to make sure we aren't adding extra values unless they are necessary here for some decision making/logic around features. cc @chance-coleman too for thoughts.

---

I see your point now! At first I wanted to use it more for the future, but I guess you're right - let's remove it and just use `tlsCertificateHeader`. If someone wants to use the Envoy Filter without actually passing a Client Certs, he can always set it to "dummy" or "ignore" and get all the things setup this way.